operator/src/main/java/oracle/kubernetes/operator/logging/OncePerMessageLoggingFilter.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at
+// http://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator.logging;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** A LoggingFilter that logs each log message, which are typically message keys, at most once */
+public class OncePerMessageLoggingFilter implements LoggingFilter {
+
+  // allow all messages to be logged when filtering is off
+  volatile boolean filtering = false;
+
+  List messagesLogged = new ArrayList();
```

Comment:
Just a minor performance note:  this could be a HashSet instead of an ArrayList.   Avoids needing to internally iterate through the whole list to implement 'contains' check...